source/retryable-reads/retryable-reads.rst

```diff
@@ -0,0 +1,644 @@
+persistent outage. In the case of a transient error, the driver will mark the
+server as "unknown" per the `SDAM
+<https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst>`__
+spec. A subsequent retry attempt will allow the driver to rediscover the primary
+within the designated server selection timeout period (30 seconds by
+default). If server selection times out during this retry attempt, we can
+reasonably assume that there is a persistent outage. In the case of a persistent
+outage, multiple retry attempts are fruitless and would waste time. See `How To
+Write Resilient MongoDB Applications
+<https://emptysqua.re/blog/how-to-write-resilient-mongodb-applications/>`__ for
+additional discussion on this strategy.
+
+Can drivers resend the same wire protocol message on retry attempts?
+--------------------------------------------------------------------
+
+No. `This is in contrast to the answer supplied in in the retryable writes
+specification. <https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#can-drivers-resend-the-same-wire-protocol-message-on-retry-attempts>`__
+However, when retryable writes were implemented, no driver actually chose to
+resend the same wire protocol message. Today, if a driver attempted to resend
+the same wire protocol message, this could violate `the rules for gossiping
+$clusterTime
+<https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#gossipping-the-cluster-time>`__:
+specifically `the rule that a driver must send the highest seen $clusterTime
+<https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#sending-the-highest-seen-cluster-time>`__.
+
+Additionally, there would be a behavioral difference between a driver resending
+the same wire protocol message and one that does not. For example, a driver
+creates a new wire protocol message could exhibit the following characteristics:
+
+1. The second attempt to send the read command could have a higher ``$clusterTime``.
+
+2. If the initial attempt failed with a server error, then the session's
+   ``operationTime`` would be advanced and the next read would include a larger
+   ``readConcern.afterClusterTime``.
+
+A driver resends the same wire protocol message would not exhibit the above
+characteristics. Thus, in order to avoid this behavioral difference and not
+violate the rules about gossiping ``$clusterTime``, drivers MUST not resend the same
+wire protocol message.
+
+Why isn't MongoDB 4.2 required?
+-------------------------------
+
+MongoDB 4.2 was initially considered as a requirement for retryable
+reads because MongoDB 4.2 implements support for "Early Failure on
+Socket Disconnect," changing the the semantics of socket disconnect to
+prevent ops from doing work that no client is interested in. This
+prevents applications from seeing degraded performance when a read is
```

Comment:
"seeing degraded performance when a read is" -> "seeing degraded performance when an expensive read is"

Reply:
I like the clarification. Fixed!